### PR TITLE
Create codeql github action

### PR DESCRIPTION
**Pull request by seanbudd**
### Summary of the issue:
CodeQL scanning was set up via the UI.
We want to configure this via YML, so we can set the branches that it runs on more easily.

### Description of development approach
Uses default template to set up python scanning on master, beta,rc 

## Summary by CodeRabbit

- **New Features**
	- Introduced a new automated workflow for CodeQL code scanning to enhance code security and quality.
	- The workflow triggers on specific events and runs on a scheduled basis to ensure continuous monitoring.
	- Supports analysis in multiple programming languages, starting with Python.

### CODE REVIEW

File: .github/workflows/codeql.yml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+name: "CodeQL code scanning"
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+    branches: [ "master" ]
```
Comment (seanbudd) [inline .github/workflows/codeql.yml:7]:
```suggestion
    branches: [ "master", "beta", "rc" ]
  pull_request:
    branches: [ "master", "beta", "rc" ]
```

File: .github/workflows/codeql.yml [hunk 2]
```diff
@@ -0,0 +1,68 @@
+name: "CodeQL code scanning"
+
+on:
+  push:
+    branches: [ "master", "beta", "rc" ]
+  pull_request:
+    branches: [ "master", "beta", "rc" ]
+  schedule:
+    - cron: '35 22 * * 3'
```
Comment (SaschaCowley) [inline .github/workflows/codeql.yml:9]: Why 22:35 on Wednesdays?

Comment (SaschaCowley) [inline .github/workflows/codeql.yml:9]: Will this run on all branches?

**Review by SaschaCowley — COMMENTED** [review]
Seems fine, I just have a couple of questions

File: .github/workflows/codeql.yml (hunk 2 above)
Comment (seanbudd) [reply .github/workflows/codeql.yml:9]: I think that's randomly generated to avoid batching on server

Comment (seanbudd) [reply .github/workflows/codeql.yml:9]: no - only master: https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#schedule

**Comment by seanbudd** [comment]
rebased to beta

File: .github/workflows/codeql.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+name: "CodeQL code scanning"
+
+on:
+  push:
+    branches: [ "master", "beta", "rc" ]
+  pull_request:
+    branches: [ "master", "beta", "rc" ]
+  schedule:
```
Comment (SaschaCowley) [inline .github/workflows/codeql.yml:8]:
```suggestion
  schedule:
    # Scheduled jobs only run on the default branch.
```

**Review by SaschaCowley — APPROVED** [review]

**Pull request merged**